File: scripts/load-v37.js
#!/usr/bin/env node

const { execSync } = require('child_process');
const path = require('path');

try {
  process.chdir('/vercel/share/v0-project');
  
  // Get all commits for the file
  const logs = execSync('git log --oneline components/report-section.tsx').toString().split('\n');
  console.log('Recent commits:');
  logs.slice(0, 10).forEach((log, i) => console.log(`${i}: ${log}`));
  
  // Load commit from 37 versions ago (approximately)
  if (logs.length > 37) {
    const targetCommit = logs[37].split(' ')[0];
    console.log(`\nLoading version at commit: ${targetCommit}`);
    execSync(`git checkout ${targetCommit} -- components/report-section.tsx`);
    console.log('Version 37 loaded successfully');
  } else {
    console.log(`Only ${logs.length} commits found. Loading the oldest available version.`);
    const oldestCommit = logs[logs.length - 1].split(' ')[0];
    execSync(`git checkout ${oldestCommit} -- components/report-section.tsx`);
  }
} catch (error) {
  console.error('Error:', error.message);
  process.exit(1);
}

File: scripts/load-v37.py
#!/usr/bin/env python3
import subprocess
import sys
import os

os.chdir('/vercel/share/v0-project')

# Try to get git log to find version 37
try:
    # First, let's see what's available
    result = subprocess.run(['git', 'log', '--oneline', '-n', '50'], capture_output=True, text=True)
    print("Recent commits:")
    print(result.stdout)
    
    # Try to checkout version 37
    checkout = subprocess.run(['git', 'checkout', 'v37', '--', 'components/report-section.tsx'], 
                            capture_output=True, text=True)
    if checkout.returncode == 0:
        print("\n✓ Successfully loaded version 37 of report-section.tsx")
    else:
        # Try HEAD~37 (37 commits back)
        checkout = subprocess.run(['git', 'checkout', 'HEAD~37', '--', 'components/report-section.tsx'], 
                                capture_output=True, text=True)
        if checkout.returncode == 0:
            print("\n✓ Successfully loaded report-section.tsx from 37 commits ago")
        else:
            print(f"Error: {checkout.stderr}")
            sys.exit(1)
            
except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
